### ISSUE TITLE
Optimize Resource Whitelist

### ISSUE THREAD

**Issue by simpleton** [shwenzhang/AndResGuard#58]
Read 'applicationId' from variant

**Comment by openproject** [comment]
You can get packageName by applicationId:
```java
andResGuard {
    def packageName = android.defaultConfig.applicationId
    whiteList = [
            packageName + ".R.drawable.icon"
    ]
}
```

**Comment by openproject** [comment]
the defaultConfig is not always right, it need better solutions!

**Comment by simpleton** [comment]
#fixed in 1.1.8 release
